File: tests/wiringPi/softPwm.py
import importlib
import pytest

m=importlib.import_module("wiringop.wiring_pi.soft_pwm")

def test_soft_pwm_create():
    _=m.soft_pwm_create(1, 1, 1)

def test_soft_pwm_write():
    _=m.soft_pwm_write(1, 1)

def test_soft_pwm_stop():
    _=m.soft_pwm_stop(1)

File: support/mock_backend/wiringPi/softPwm.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <string.h>
#include "wiringPi/softPwm.h"
int softPwmCreate(int pin, int value, int range) { (void)pin;(void)value;(void)range;return 1; }
void softPwmWrite(int pin, int value) { (void)pin;(void)value; }
void softPwmStop(int pin) { (void)pin; }
